File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        int n=grid.length;
        int m=grid[0].length;
        int arr[]=new int[n*m];
        int index=0;

        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                arr[index++]=grid[i][j];
            }
        }

        int rem=arr[0]%x;

        for(int ele:arr){
            if((ele%x)!=rem) return -1;
        }

        Arrays.sort(arr);

        int median=arr[arr.length/2];
        int ans=0;
        for(int ele:arr){
            ans+=Math.abs(ele-median)/x;
        }
        return ans;
    }
}

File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int>arr;

        for(auto row:grid){
            for(auto ele:row){
                arr.emplace_back(ele);
            }
        }

        int rem = arr[0] % x;
        for(int val : arr){
            if(val % x != rem) return -1;
        }

        sort(arr.begin(),arr.end());
        int ans=0;

        int median=arr[arr.size()/2];
        for(auto val:arr){
            ans+=abs(val-median)/x;
        }
        return ans;
    }
};